src/EnergyPlus/DataSurfaces.hh

```diff
@@ -65,6 +65,33 @@
 #include <EnergyPlus/EnergyPlus.hh>
 #include <EnergyPlus/Shape.hh>
 
+
+#define BITF(B) (1 << (int(B)))
+#define BITF_TEST_ANY(V, B) (((V) & (B)) != 0)
+
+// IS_SHADED is the flag to indicate window has no shading device or shading device is off, and no daylight glare control
+// original expression: SHADE_FLAG == ShadeOff || SHADE_FLAG == ShadeOff
+#define NOT_SHADED(SHADE_FLAG) BITF_TEST_ANY(BITF(SHADE_FLAG), BITF(WinShadingType::NoShade) | BITF(WinShadingType::ShadeOff))
+
+// IS_SHADED is the flag to indicate window has shade on or temporarily off but may be triggered on later to control daylight glare
+// original expression: SHADE_FLAG > ShadeOff
+#define IS_SHADED(SHADE_FLAG) !NOT_SHADED(SHADE_FLAG)
+
+// IS_SHADED_NO_GLARE is the flag to indicate window has shade and no daylight glare control
+// original expression: IntShade <= SHADE_FLAG <= BGBlind
+#define IS_SHADED_NO_GLARE_CTRL(SHADE_FLAG) BITF_TEST_ANY(BITF(SHADE_FLAG), \
+                                                     BITF(WinShadingType::IntShade) | BITF(WinShadingType::SwitchableGlazing) | BITF(WinShadingType::ExtShade) | BITF(WinShadingType::ExtScreen) |\
+                                                     BITF(WinShadingType::IntBlind) | BITF(WinShadingType::ExtBlind) | BITF(WinShadingType::BGShade) | BITF(WinShadingType::BGBlind))
+
+// ANY_SHADE: if SHADE_FLAG is any of the shading types including interior, exterior or between glass shades
+#define ANY_SHADE(SHADE_FLAG) BITF_TEST_ANY(BITF(SHADE_FLAG), BITF(WinShadingType::IntShade) | BITF(WinShadingType::ExtShade) | BITF(WinShadingType::BGShade))
+#define ANY_SHADE_SCREEN(SHADE_FLAG) BITF_TEST_ANY(BITF(SHADE_FLAG), BITF(WinShadingType::IntShade) | BITF(WinShadingType::ExtShade) | BITF(WinShadingType::BGShade) | BITF(WinShadingType::ExtScreen))
+#define ANY_BLIND(SHADE_FLAG) BITF_TEST_ANY(BITF(SHADE_FLAG), BITF(WinShadingType::IntBlind) | BITF(WinShadingType::ExtBlind) | BITF(WinShadingType::BGBlind))
+#define ANY_INTERIOR_SHADE_BLIND(SHADE_FLAG) BITF_TEST_ANY(BITF(SHADE_FLAG), BITF(WinShadingType::IntShade) | BITF(WinShadingType::IntBlind))
+#define ANY_EXTERIOR_SHADE_BLIND_SCREEN(SHADE_FLAG) BITF_TEST_ANY(BITF(SHADE_FLAG), BITF(WinShadingType::ExtShade) | BITF(WinShadingType::ExtBlind) | BITF(WinShadingType::ExtScreen))
+#define ANY_BETWEENGLASS_SHADE_BLIND(SHADE_FLAG) BITF_TEST_ANY(BITF(SHADE_FLAG), BITF(WinShadingType::BGShade) | BITF(WinShadingType::BGBlind))
```

Comment:
With hindsight, I wonder if these could be constexpr functions that still achieve the performance gains but allow more insight with IDEs.  Nothing to go back and change here for now, just something to think about before we spread this pattern around other places.

Reply:
The `SHADE_FLAG` argument itself is not `constexpr`, so a `constexpr` function will not help or at least cannot be the entire solution. If you want something that is closer to the original code and maybe more transparent at the call site, you could use something like:

`#define BITF_TEST_ANY3(FLAG, B1, B2, B3) BITF_TEST_ANY(BITF(FLAG), BITF(B1) | BITF(B2) | BITF(B3))`

`if (BITF_TEST_ANY3(ShadeFlag, WinShadingType::IntBlind, WinShadingType::ExtBlind, WinShadingType::BGBlind))`

Is that better than:

`if (TEST_ANY_BLIND(ShadeFlag))`

? I don't know, is it?